server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1736,7 +1737,10 @@ public boolean storagePoolHasEnoughSpace(List<Volume> volumes, StoragePool pool)
                     allocatedSizeWithtemplate = _capacityMgr.getAllocatedPoolCapacity(poolVO, tmpl);
                 }
             }
-            if (volumeVO.getState() != Volume.State.Ready) {
+            // A ready state volume is already allocated in a pool. so the asking size is zero for it.
+            // In case the volume is moving across pools or is not ready yet, the asking size has to be computed
+            s_logger.debug("pool id for the volume with id: " + volumeVO.getId() + " is: " + volumeVO.getPoolId());
```

Comment:
Could you make this line a bit more descriptive? To me it doesn't seem like a very useful line when running in debug mode.

Also, do we want the numeric IDs here?